octavia-cli/octavia_cli/headers.py

```diff
@@ -0,0 +1,86 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from dataclasses import dataclass
+from typing import Optional, List
+
+import yaml
+
+from .apply.yaml_loaders import EnvVarLoader
+
+
+API_HEADERS_YAML_FILE_INSTRUCTION = """
+headers:
+  - name: Authorization
+    value: Basic dXNlcjpwYXNzd29yZA==
+  - name: Content-Type
+    value: application/json
+""".strip()
+
+
+@dataclass
+class ApplicationHeader:
+    name: Optional[str]
+    value: Optional[str]
+
+    @property
+    def is_valid(self):
+        return all([self.name, self.value])
+
+    def clean(self):
+        return ApplicationHeader(
+            name=self.name.strip(),
+            value=self.value.strip()
+        )
+
+
+def read_yaml_header_configuration(header_configuration_path: str) -> List[ApplicationHeader]:
+    with open(header_configuration_path) as file:
+        content = yaml.load(file, EnvVarLoader)
+
+    headers = content.get("headers", None)
+
+    if not headers:
+        raise AttributeError("""
+        Please provide valid configuration file, like below
+        headers:
+          - name: Authorization
+            value: Basic dXNlcjpwYXNzd29yZA==
+          - name: Content-Type
+            value: application/json
+        """)
+
+    application_headers = [
+        ApplicationHeader(header_pair.get("name", None), header_pair.get("value")) for header_pair in headers
+    ]
+
+    valid_headers = [
+        app_header.clean() for app_header in application_headers if app_header.is_valid
+    ]
```

Comment:
IMO this cleaning should happen on initialization of the dataclass. https://docs.python.org/3/library/dataclasses.html#post-init-processing